src/api_module.js

```diff
@@ -8,194 +8,301 @@ const fetchRetry = useFetchRetry(fetch, {
+          "x-device-id": auth.device_id,
+        },
+      });
+    } catch (error) {
+      const reason = {};
+      reason[error.constructor.name] = [error.message];
+      throw reason;
     }
-
-    if (routes.includes("indexWithScope")) {
-      this.indexGenerator = function (auth, scope = new Scope()) {
-        return this.#internalIndexGenerator(auth, scope);
-      };
+    const result = await response.json();
+    const loaded = new Date();
+    for (let obj in result) {
+      result[obj].loaded = loaded;
     }
-
-    if (routes.includes("create")) {
-      this.create = async function (auth, object) {
-        const request = new Request(`${this.path}`, {
-          method: "POST",
-          headers: {
-            "content-type": "application/json",
-            "x-secret": auth?.secret,
-            "x-device-id": auth?.device_id,
-          },
-          body: JSON.stringify(object),
-        });
-        return await this._resolveRequest(request);
-      };
+    if (response.ok && result) {
+      if (response.headers.get("x-total-pages") === page.toString()) {
+        return result;
+      } else {
+        yield result;
+      }
+    } else {
+      throw result;
     }
 
-    if (routes.includes("read")) {
-      this.read = async function (auth, id, retryOptions = {}) {
-        const request = new Request(`${this.path}/${id}`, {
-          ...retryOptions,
-          method: "GET",
-          headers: {
-            "x-secret": auth.secret,
-            "x-device-id": auth.device_id,
-          },
-        });
-        return await this._resolveRequest(request);
-      };
-    }
+    page++;
+  }
+}
 
-    if (routes.includes("update")) {
-      this.update = async function (auth, id, object) {
-        const request = new Request(`${this.path}/${id}`, {
-          method: "PATCH",
-          headers: {
-            "content-type": "application/json",
-            "x-secret": auth.secret,
-            "x-device-id": auth.device_id,
-          },
-          body: JSON.stringify(object),
-        });
-        return await this._resolveRequest(request);
-      };
-    }
+async function httpGet(path, auth, retryOptions) {
+  const request = new Request(`${this.path}`, {
+    ...retryOptions,
+    method: "GET",
+    headers: {
+      "x-secret": auth.secret,
+      "x-device-id": auth.device_id,
+    },
+  });
+  return await resolve(request);
+}
 
-    if (routes.includes("destroy")) {
-      this.destroy = async function (auth, id) {
-        const request = new Request(`${this.path}/${id}`, {
-          method: "DELETE",
-          headers: {
-            "x-secret": auth.secret,
-            "x-device-id": auth.device_id,
-          },
-        });
-        return await this._resolveRequest(request);
-      };
-    }
+async function httpPost(path, auth, data) {
+  const request = new Request(path, {
+    method: "POST",
+    headers: {
+      "content-type": "application/json",
+      "x-secret": auth.secret,
```

Comment:
Should be `auth?.secret` no? We don't require when creating auth tokens

Reply:
I opted to special case this in the auth tokens module, passing an empty object. This will result in the values here being undefined, thus the headers not being sent.